src/Microsoft.DotNet.Interactive/PackageRestoreContext.cs

```diff
@@ -234,6 +233,11 @@ public async Task<PackageRestoreResult> RestoreAsync()
                     succeeded: false,
                     requestedPackages: newlyRequested,
                     errors: errors);
+
+                foreach(var r in newlyRequested)
+                {
+                    _requestedPackageReferences.TryRemove(r.PackageName.ToLower(CultureInfo.InvariantCulture), out var _);
+                }
```

Comment:
This looks like a breaking change. We don't want to remove the requested packages, we just want to not re-request them. The information is useful in other ways, e.g. we know someone requested `FluentAssertions,*` and received `FluentAssertions,5.7.0`.

Reply:
@jonsequitur, it's fine this is the failed code path, as soon as the developer re-submits the script code _requestedPackageReferences will be refreshed.

Reply:
Why not marking the packages as 
- pending
- succeeded
- failed
So we avoid requesting again anything that worked or failed (maybe keep the failure reason too?)

Reply:
@colombod, failure may be due to a point in time effect such as a network latency problem or it may succeed by adding a #i "nuget:something" or a #i may fail, but succeed after the developer created the actual directory.

Reply:
Deliberately re-requesting a failed package is a great scenario that deserves its own test, e.g. failing, adding a `#i`, trying again, and succeeding.